CRM/Chreports/Reports/BaseReport.php

```diff
@@ -398,47 +257,87 @@ public function getReportName(): string {
+    // public function isMonthlyYearlyReport(): bool {
+    //     if(($this->_settings['period'] == 'monthly' && $this->_settings['name'] !== 'contrib_monthly_fiscal_year') || $this->_settings['period'] == 'yearly')
+    //     {
+    //         $this->_limit = '';
+    //         return true; 
+    //     }
+    //     return false;
+    // }
+
+    public function isPeriodicSummary (): bool {
+        if($this->_settings['template'] == 'chreports/contrib_period_summary')
         {
             $this->_limit = '';
             return true; 
         }
         return false;
     }
 
-    //check if report is repeat contribution report
+    //Retrieve Repeat Contributions Report
     public function isRepeatContributionReport(): bool {
-        if($this->_settings['name'] == 'repeat_contributions_detailed' && $this->_settings['entity'] == 'contact')
-        {
-            $this->setEntity($this->_settings['entity']);
-            return true; 
-        }
-        return false;
+        $status = ($this->_settings['template'] == 'chreports/contrib_period_compare') ? true : false;
+        return $status;
     }
     //check if report is opportunity report
     public function isOpportunityReport(): bool {
-        if($this->_settings['name'] == 'opportunity_details' && $this->_settings['entity'] == 'grant')
-        {
-            return true; 
-        }
-        return false;
+        $status = ($this->_settings['template'] == 'chreports/opportunity_detailed') ? true : false;
+        return $status;
     }
     //check if report is opportunity report
     public function isLYBNTSYBNTReport(): bool {
-        if($this->_settings['name'] == 'contrib_sybunt' || $this->_settings['name'] == 'contrib_lybunt')
-        {
-            return true; 
-        }
-        return false;
+        $status = (in_array($this->_settings['template'], ['chreports/lybunt' , 'chreports/sybunt'])) ? true : false;
+        return $status;
+    }
+
+    //check if report is TopDonor report
+    public function isTopDonorReport(): bool {
+        $status = ($this->_settings['template'] == 'chreports/top_donors') ? true : false;
+        return $status;
+    }
+ 
+    //Retrieve GL Accountand Payment Method Reconciliation Report
+    public function isGLAccountandPaymentMethodReconciliationReport()
+    {
+        $status = ($this->_settings['template'] == 'chreports/contrib_glaccount') ? true : false;
+        return $status;
+    }
+   
+    //Retrieve Recurring Contribution Report
+    public function isRecurringContributionReport()
+    {
+        $status = ($this->_settings['entity'] == 'contact' && $this->_settings['template'] == 'chreports/contrib_detailed') ? true : false;
+        return $status;
+    }
+
+    public function hasMonthlyBreakdown (): bool {
+        $status = $this->_settings['period'] == 'monthly' ? true : false;
```

Comment:
We can simplify a bit.

```suggestion
       return $this->_settings['period'] == 'monthly';
```